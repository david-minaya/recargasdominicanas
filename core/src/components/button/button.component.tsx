import clsx from 'clsx';
import React, { MouseEvent } from 'react';
import { Icon } from '../icon/icon.component';
import { Text } from '../text/text.component';
import { Style, mergeStyle } from './button.module.css';

export interface Props {
  id?: string;
  text?: string;
  icon?: string;
  rightIcon?: string;
  hidden?: boolean;
  disabled?: boolean;
  style?: Style;
  onClick?: (event?: MouseEvent<HTMLButtonElement>) => void;
  onMouseDown?: (event?: MouseEvent<HTMLButtonElement>) => void;
}

export function Button(props: Props) {

  const {
    id,
    text = 'Button',
    icon,
    rightIcon,
    disabled = false,
    hidden = false,
    style: customStyle,
    onClick,
    onMouseDown
  } = props;

  const style = mergeStyle(customStyle);

  function handleClick(event?: MouseEvent<HTMLButtonElement>) {
    if (!disabled && !hidden && onClick) {
      onClick(event);
    }
  }

  function handleMouseDown(event?: MouseEvent<HTMLButtonElement>) {
    if (!disabled && !hidden && onMouseDown) {
      onMouseDown(event);
    }
  }

  return (
    <button
      id={id}
      onClick={handleClick}
      onMouseDown={handleMouseDown}
      className={clsx(
        style.base,
        disabled && style.disabled,
        hidden && style.hidden
      )}>
      {icon &&
        <Icon className={style.icon} icon={icon}/>
      }
      <Text className={style.text} text={text}/>
      {rightIcon &&
        <Icon className={style.rightIcon} icon={rightIcon}/>
      }
    </button>
  );
}
