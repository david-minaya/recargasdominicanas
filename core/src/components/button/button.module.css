.base {
  width: fit-content;
  height: 36px;
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  white-space: nowrap;
  user-select: none;
  outline: none;
  border: none;
  background-color: white;
  border-radius: 4px;
  padding: 0px 16px;
  -webkit-tap-highlight-color: transparent;
}

@media (pointer: fine) {

  .base:hover {
    background-color: rgb(241 38 38 / 0.03);
  }
}

.base:active {
  background-color: rgb(241 38 38 / 0.06);
}

.hidden {
  display: none;
}

.icon {
  cursor: pointer;
  color: var(--accent-color);
  margin-right: 16px;
}

.rightIcon {
  cursor: pointer;
  color: var(--accent-color);
  margin-left: 16px;
}

.text {
  font-size: 15px;
  font-weight: 500;
  font-family: Roboto;
  color: var(--accent-color);
  user-select: none;
}

.disabled .text {
  color: lightgray;
}

.disabled.base {
  cursor: default;
}

.disabled .icon {
  color: lightgray;
}
