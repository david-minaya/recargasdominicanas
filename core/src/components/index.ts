export * from './icon/icon.component';
export * from './option-icon/option-icon.component';
export * from './text/text.component';
export * from './image/image.component';
export * from './title/title.component';
export * from './outline-card/outline-card.component';
export * from './field-error/field-error.component';
export * from './text-field/text-field.component';
export * from './text-area/text-area.component';
export * from './select-option/select-option.component';
export * from './select/select.component';
export * from './autocomplete/autocomplete.component';
export * from './button/button.component';
export * from './fill-button/fill-button.component';
export * from './outline-button/outline-button.component';
export * from './error/error.component';
export * from './outline-item/outline-item.component';
export * from './detail-item/detail-item.component';
export * from './setting-item/setting-item.component';
export * from './transaction-item/transaction-item.component';
export * from './product-item/product-item.component';
export * from './option/option.component';
export * from './sales-report/sales-report.component';
export * from './brand/brand.component';
export * from './toolbar-title/toolbar-title.component';
export * from './toolbar/toolbar.component';
export * from './header/header.component';
export * from './navbar/navbar.component';
export * from './navbar-option/navbar-option.component';
export * from './menu/menu.component';
export * from './tab/tab.component';
export * from './tab-bar/tab-bar.component';
export * from './modal/modal.component';
export * from './modal-toolbar/modal-toolbar.component';
export * from './modal-content/modal-content.component';
export * from './modal-options/modal-options.component';
export * from './modal-actions/modal-actions.component';
export * from './base-modal/base-modal.component';
export * from './loading-modal/loading-modal.component';
export * from './receipt/receipt.component';
export * from './receipt-modal/receipt-modal.component';
export * from './switch/switch.component';
export * from './switch-case/switch-case.component';
export * from './cancel-transaction-modal/cancel-transaction-modal.component';
export * from './base-cell/base-cell.component';
export * from './cell/cell.component';
export * from './cell-button/cell-button.component';
export * from './cell-icon/cell-icon.component';
export * from './cell-image/cell-image.component';
export * from './outline-table/outline-table.component';
export * from './page-container/page-container.component';
export * from './page-content/page-content.component';
export * from './page-toolbar/page-toolbar.component';
export * from './table/table.component';
export * from './table-header/table-header.component';
export * from './table-row/table-row.component';
export * from './transaction-modal/transaction-modal.component';
export * from './confirm-transaction-modal/confirm-transaction-modal.component';
export * from './error-transaction-modal/error-transaction-modal.component';
export * from './drawer/drawer.component';
export * from './drawer-header/drawer-header.component';
export * from './drawer-item/drawer-item.component';
export * from './protected-route/protected-route.component';
export * from './float-button/float-button.component';
export * from './datetime/datetime.component';
export * from './app-toolbar/app-toolbar.component';
export * from './sales-report-modal/sales-report-modal.component';
export * from './toast/toast.component';
export * from './error-page/error-page.component';
export * from './progress/progress.component';
export * from './indeterminate-progress-bar/indeterminate-progress-bar.component';
export * from './alert/alert.component';
export * from './pagination/pagination.component';
export * from './text-item/text-item.component';
