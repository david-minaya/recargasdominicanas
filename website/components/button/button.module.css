.base {
  width: 180px;
  height: 40px;
  cursor: pointer;
  font-size: 16px;
  font-weight: 500;
  font-family: Roboto;
  color: var(--accent-color);
  background-color: white;
  outline: none;
  border: none;
  padding: 0px;
  -webkit-tap-highlight-color: transparent;
}

.disabled {
  color: lightgray;
}

.hidden {
  display: none;
}
