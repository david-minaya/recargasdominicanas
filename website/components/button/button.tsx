import React from 'react';
import clsx from 'clsx';
import { mergeStyle } from './button.module.css';

interface props {
  id?: string;
  text?: string;
  style?: { base?: string, disabled?: string, hidden?: string };
  hidden?: boolean;
  disabled?: boolean;
  onClick?: () => void;
}

export function Button(props: props) {

  const {
    id,
    style: customStyle,
    text = 'Button',
    disabled = false,
    hidden = false,
    onClick
  } = props;

  const style = mergeStyle(customStyle);

  function handleClick() {
    if (!disabled && !hidden) {
      onClick();
    }
  }

  return (
    <button 
      id={id}
      className={clsx(
        style.base, 
        disabled && style.disabled,
        hidden && style.hidden 
      )}
      onClick={handleClick}>
      {text}
    </button>
  );
}
