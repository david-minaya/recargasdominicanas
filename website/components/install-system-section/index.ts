export * from './install-system-section';
