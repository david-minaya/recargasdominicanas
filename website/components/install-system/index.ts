export { InstallSystem } from './install-system';
