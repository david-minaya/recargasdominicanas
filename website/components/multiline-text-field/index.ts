export * from './multiline-text-field';
