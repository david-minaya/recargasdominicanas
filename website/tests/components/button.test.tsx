import React from 'react';
import { shallow } from 'enzyme';
import { Button } from '../../components/button';

it('click TextButton', () => {
  const handleClick = jest.fn();
  const wrapper = shallow(<Button onClick={handleClick}/>);
  wrapper.simulate('click');
  expect(handleClick).toHaveBeenCalled();
});

describe('render TextButton', () => {

  it('render with the default text', () => {
    const wrapper = shallow(<Button/>);
    expect(wrapper.text()).toBe('Button');
  });
  
  it('render with custom text', () => {
    const wrapper = shallow(<Button text='Text Button'/>);
    expect(wrapper.text()).toBe('Text Button');
  });
});

describe('TextButton snapshots', () => {
  
  it('snapshot with the default text', () => {
    const wrapper = shallow(<Button/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('snapshot with custom text', () => {
    const wrapper = shallow(<Button text='Text Button'/>);
    expect(wrapper).toMatchSnapshot();
  });
});
