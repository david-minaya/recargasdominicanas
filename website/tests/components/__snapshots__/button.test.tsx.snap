// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextButton snapshots snapshot with custom text 1`] = `
<button
  className="button"
>
  Text Button
</button>
`;

exports[`TextButton snapshots snapshot with the default text 1`] = `
<button
  className="button"
>
  Button
</button>
`;
